src/main.rs

```diff
@@ -73,9 +76,9 @@ pub fn load_state(filepath: &str) -> CrunchyState {
 // doing too many things. Especially, when I'd like to add here some other stuff like peer sharing it would
 // be too messy. It should be re-designed and divided into smaller functions with appropriate names and
 // functionalities (like computing graphs, counting factors, geolocalization etc).
-async fn write_state(infile: &str, outfile: &str, cachefile: &str) {
-    let mut geo_cache = GeoIPCache::new(Path::new(cachefile));
-    let response = load_response(infile);
+async fn write_state(config: &CrunchyConfiguration) {
```

Comment:
The whole geo_cache struc could be setup in a separate function

Reply:
Totally agree. 
And as stated in the TODO: https://github.com/runziggurat/crunchy/blob/d29427f6f202ae394e1c32cabfb187691e067585/src/main.rs#L72 it should be done at some point of time.